### ISSUE TITLE
[Dove] Client side hooks for custom methods

### ISSUE THREAD

**Issue by joezappie** [feathersjs/feathers#2619]
### Steps to reproduce

I'm using the new custom methods in Dove and while custom hooks are working for backend methods, they do not appear to work client side Here is an example:

```
const service = client.service('workorders');

// Init custom methods
service.methods('addToInvoice');

// Setup hooks
service.hooks({
  before: {
    all: [common.addOrg],
    addToInvoice: [ addRefType ],
  },
};
```

In this example I get the error:
`Uncaught Error: 'addToInvoice' is not a valid hook method`.

I've also notice that the `all` hook is not called for custom methods as well.

I'm sure you already have plans for this but figured having an issue to track it would be nice.

### Expected behavior
Hook should run when that custom method is called.

### Actual behavior
Errors saying its not a valid hook method. `all` method not ran either.

### System configuration

Tell us about the applicable parts of your setup.

**Module versions** (especially the part that's not working):
v5.0.0-pre.18

**NodeJS version**:
v16.13.1

**Operating System**:
Windows 10

**Browser Version**:
[Version 1.37.116 Chromium: 100.0.4896.127 (Official Build) (64-bit)](https://brave.com/latest/)

**React Native Version**:
N/A

**Module Loader**:
Webpack

**Comment by daffl** [comment]
I believe it does work already with new async hook functions (after calling `service.methods('addToInvoice')`):

```js
service.hooks({
  addToInvoice: [ async (context, next) => {
    // Do stuff before
    await next();
    // Do stuff after
  }]
};
````

But I'm kind of not really happy how client side custom methods are initialised at the moment anyway. I'm thinking it should work essentially the same as on the server via the `options` in `app.use('name', service, options)` - in which case proper hooks will be available, too:

```ts
import { feathers, CustomMethod } from '@feathersjs/feathers';
import socketio, { SocketService } from '@feathersjs/socketio-client';
import io from 'socket.io-client';

type ServiceTypes = {
  myservice: SocketService & CustomMethod<'myCustomMethod'>
}

const socket = io('http://api.feathersjs.com');
const app = feathers<ServiceTypes>();
const connection = socketio(socket)

// Set up Socket.io client with the socket
app.configure(connection);

client.use('myservice', connection.service('myservice'), {
  methods: [ 'find', 'myCustomMethod' ]
})

// Then it can be used like other service methods
client.service('myservice').myCustomMethod(data, params);
```

**Comment by joezappie** [comment]
I tried making it async but no luck.

I did just realized that calling `service.method()` to early causes this. Wrapping that and the `service.hooks()` call in a timeout makes it work:

```
const service = client.service('workorders');

setTimeout(() => {
  service.methods('addToInvoice');

  service.hooks({
    before: {
      all: [common.addOrg],
      addToInvoice: [ addRefType ],
    },
  };
}, 100);
```

Both methods() and hooks() has to be deferred for it to work for me.

Obviously timeouts are not a good solution here but it must be getting overwritten as the service gets setup?

Also kind of weird that the `all` hook wasn't getting called on the custom method but  I was able to call the method. So seems like its initialized but at the same time not.

That said, I can easily get around the issue for now since it sounds like you want to tear into the whole client a lot deeper. My hook was just to force a query parameter but I can just manually do that for now.
